src/cmap/connect.ts

```diff
@@ -125,6 +125,15 @@ function performInitialHandshake(
         return;
       }
 
+      if (options.loadBalanced && !response.serviceId) {
```

Comment:
Throw an error if in LB mode and no serviceId in the handshake response per the spec.